### ISSUE TITLE
[BUG] SSR does not work in `/map` and `/dashboards/electricity` pages

### ISSUE THREAD

**Issue by JinIgarashi** [UNDP-Data/geohub#1939]
### Bug Description
If SSR is turned on, the pages which uses MenuControl does not work properly. It actually works from other pages' link, but it does not work when it is reloaded in browser or direct link.

The following error log is in nodejs server

```shell
2023-09-27T17:40:32.203401831Z ReferenceError: require is not defined in ES module scope, you can use import instead
2023-09-27T17:40:32.203447531Z This file is being treated as an ES module because it has a '.js' file extension and '/home/site/wwwroot/package.json' contains "type": "module". To treat it as a CommonJS script, rename it to use the '.cjs' file extension.
2023-09-27T17:40:32.203455232Z     at file:///home/site/wwwroot/server/chunks/MenuControl-eea18ba8.js:3854:16
2023-09-27T17:40:32.203460632Z     at ModuleJob.run (node:internal/modules/esm/module_job:194:25)
```

Everything works well in local, this bug is only there in produciton.

### Bug Reproduction
remove following code from +page.ts to turn on SSR, then deploy to produciton.

```js
export const csr = true;
export const ssr = false;
```

### Bug Impact
SEO cannot work for `/map` and `/dashboards/electricity` pages.

### Bug System Information

* OS: [e.g. Windows 10]
* Browser: [e.g. Google Chrome, Firefox]

┆Issue is synchronized with this [Wrike task](https://www.wrike.com/open.htm?id=1215123914) by [Unito](https://www.unito.io)

**Comment by JinIgarashi** [comment]
I tried SSR on Electricity Dashboard, got the below error.
Looks like vega having issue on SSR

```
ReferenceError: require is not defined in ES module scope, you can use import instead
This file is being treated as an ES module because it has a '.js' file extension and '/geohub/package.json' contains "type": "module". To treat it as a CommonJS script, rename it to use the '.cjs' file extension.
    at file:///geohub/server/chunks/vega-embed.module-e77d3307.js:3878:16
    at ModuleJob.run (node:internal/modules/esm/module_job:193:25)
```

**Comment by JinIgarashi** [comment]
fixed by #2930
